src/libAtomVM/nifs.c

```diff
@@ -3803,6 +3810,52 @@ static term nif_atomvm_read_priv(Context *ctx, int argc, term argv[])
     return result;
 }
 
+static term nif_atomvm_posix_clock_settime(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc);
+
+    VALIDATE_VALUE(argv[0], term_is_atom);
+    VALIDATE_VALUE(argv[1], term_is_any_integer);
+
+#ifdef HAVE_CLOCK_SETTIME
+
+    term unit = argv[0];
+    avm_int64_t value = term_maybe_unbox_int64(argv[1]);
+
+    struct timespec tp;
+    if (unit == SECOND_ATOM) {
+        tp.tv_sec = value;
+        tp.tv_nsec = 0L;
+
+    } else if (unit == MILLISECOND_ATOM) {
+        tp.tv_sec = value / 1000;
+        tp.tv_nsec = (value % 1000) * 1000000;
+
+    } else if (unit == MICROSECOND_ATOM) {
+        tp.tv_sec = value / 1000000;
+        tp.tv_nsec = (value % 1000000) * 1000;
+
+    } else {
+        RAISE_ERROR(BADARG_ATOM);
+    }
+
+    int res = clock_settime(CLOCK_REALTIME, &tp);
+    if (res != 0) {
+        if (UNLIKELY(memory_ensure_free(ctx, TUPLE_SIZE(2)) != MEMORY_GC_OK)) {
+            RAISE_ERROR(OUT_OF_MEMORY_ATOM);
+        }
+        term error = term_alloc_tuple(2, &ctx->heap);
+        term_put_tuple_element(error, 0, ERROR_ATOM);
+        term_put_tuple_element(error, 1, posix_errno_to_term(errno, ctx->global));
+        RAISE_ERROR(error);
```

Comment:
This is the `{error, Reason}` I referred to above in atomvm.erl...

Reply:
That’s raising an error, not returning one.

Reply:
Now we could change that to a return value, I suppose.  It might make more sense.  Let me think about it.

Reply:
Derp. I just realized this is raised not returned. Ignore all previous comments about this.

Reply:
I think your intuition was right — better to return an error with the POSIX reason, rather than raise an error.